lib/rs/src/transport/mod.rs

```diff
@@ -20,6 +20,17 @@
 use std::io::prelude::*;
 
 pub mod tcp_transport;
+pub mod server;
+
+pub trait TransportFactory<T: Transport> {
+    fn new_protocol(&self) -> T;
```

Comment:
Ah, I actually need to remove this factory. As it is, it's useless, because the `SimpleServer` constructs transports with `accept`. This was meant to wrap transports with other transports, but that's not something we should be worrying about. When we add support for separate input and output transports, we can add this back in then.